source/js/components/petition/petition.jsx

```diff
@@ -116,26 +117,24 @@ export default class Petition extends React.Component {
     );
   }
 
-  fbButtonClicked() {
-    this.shareButtonClicked(`#share-progress-fb`);
-  }
-
-  twButtonClicked() {
-    this.shareButtonClicked(`#share-progress-tw`);
-  }
-
-  emButtonClicked() {
-    this.shareButtonClicked(`#share-progress-em`);
-  }
-
-  shareButtonClicked(id) {
+  shareButtonClicked(event, shareProgressButtonId) {
     ReactGA.event({
       category: `petition`,
       action: `share tap`,
       label: `${document.title} - share tap`
     });
 
-    document.querySelector(id + ` a`).click();
+    if (shareProgressButtonId) {
+      let shareProgressButton = document.querySelector(
+        `#${shareProgressButtonId} a`
+      );
+
+      if (shareProgressButton) {
```

Comment:
Is there a reason we need this additional `if` statement? It seems like the only way to step into the initial `if` statement on L127 is if there _is_ a `shareProgressButtonId` which means the only options available anyways would be the 3 share buttons that have the id parameter in `onClick`. So, for example we could just jump right into `shareProgressButton.click();` on L132.

Let me know if I'm missing something...it's almost end of day for me 😅